tool/release/createnotes/Note.kt

```diff
@@ -0,0 +1,145 @@
+
+    private constructor(title: String, goal: String?, type: Type) {
+        this.title = title
+        this.goal = goal
+        this.type = type
+    }
+
+    fun toMarkdown():  String {
+        return "- **$title**\n  ${goal ?: ""}"
+    }
+
+    companion object {
+        fun fromGithubPR(pr: JsonObject): Note {
+            return Note(
+                title = pr.get("title").asString(),
+                goal = getPRGoal(pr.get("body").asString()),
+                type = getPRType(pr.get("labels").asArray())
+            )
+        }
+
+        private fun getPRType(labels: JsonArray): Type {
+            val types = labels
+                .map { e -> e.asObject().get("name").asString() }
+                .filter { e -> e.startsWith(labelPrefix) }
+            val type =
+                when {
+                    types.contains(labelFeature) -> Type.FEATURE
+                    types.contains(labelBug) -> Type.BUG
+                    types.contains(labelRefactor) -> Type.REFACTOR
+                    else -> Type.OTHER
+                }
+            return type
+        }
+
+        private fun getPRGoal(description: String): String {
+            val goal = StringBuilder()
+            var header = 0
+            for (line in description.lines()) {
+                if (line.startsWith("##")) {
+                    header += 1
+                } else if (header == 1) {
+                    goal.append(line)
+                } else if (header > 1) {
+                    break
+                }
+            }
+            return goal.toString()
+        }
+
+        fun fromGithubCommit(commit: JsonObject): Note {
+            return Note(commit.get("message").asString().lines().first(), null, Type.OTHER)
+        }
+
+        fun toMarkdown(notes: List<Note>): String {
+            val features = mutableListOf<Note>()
+            val bugs = mutableListOf<Note>()
+            val refactors = mutableListOf<Note>()
+            val others = mutableListOf<Note>()
+
+            notes.forEach { note ->
+                when (note.type) {
+                    Type.FEATURE -> features.add(note)
+                    Type.BUG -> bugs.add(note)
+                    Type.REFACTOR -> refactors.add(note)
+                    else -> others.add(note)
+                }
+            }
+
+            return """
+## New Features
+${features.map(Note::toMarkdown).joinToString("\n")}
+
+## Bugs Fixed
+${bugs.map(Note::toMarkdown).joinToString("\n")}
+
+## Code Refactors
+${refactors.map(Note::toMarkdown).joinToString("\n")}
+
+## Other Improvements
+${others.map(Note::toMarkdown).joinToString("\n")}
+    """
+        }
+    }
+}
+
+fun collectNotes(org: String, repo: String, commits: List<String>, githubToken: String): List<Note> {
+    return commits.flatMap { commit ->
+        val response = httpGet("$github/repos/$org/$repo/commits/$commit/pulls", githubToken)
+        val body = Json.parse(String(response.content.readBytes()))
```

Comment:
@lolski 

```suggestion
        val body = Json.parse(response.parseAsString())
```

Reply:
Fixed in 8936646e3586dcfab42e22832c73f7db8e756a65.